### PR TITLE
interfaces: add password-manager-service implicit interface

### PR DESCRIPTION
**Pull request by mvo5**
A careful review from the security team is appreciated here. Also this was only tested with the `secret-tool` so far, no real world application yet.

LP: #1653769

### CODE REVIEW

File: interfaces/builtin/password_manager_service.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    interface=org.freedesktop.DBus.Properties
+    member="{Get,GetAll,PropertiesChanged}"
+    peer=(label=unconfined),
+
+dbus (receive, send)
+    bus=session
+    interface=org.freedesktop.Secret.*
+    path=/org/freedesktop/secrets{,/**}
+    peer=(label=unconfined),
+
```
Comment (zyga) [inline interfaces/builtin/password_manager_service.go]: Could you please remove those extra newlines here?

**Comment by mvo5** [comment]
Two comments from zyga: 
* apt-get install is not suitable in the cross distro world anymore ;)
* document interface in the forum and link from interface to it.

File: tests/main/interfaces-password-manager-service/task.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+restore: |
+    apt-get purge -y --auto-remove gnome-keyring
+
+execute: |
+    CONNECTED_PATTERN=":password-manager-service +test-snapd-password-manager-consumer"
+    DISCONNECTED_PATTERN="(?s).*?\n- +test-snapd-password-manager-consumer:password-manager-service"
+
+    echo "Ensure things run"
+    eval $(dbus-launch --sh-syntax)
```
Comment (zyga) [inline tests/main/interfaces-password-manager-service/task.yaml]: Do you have to kill this dbus process later?

File: tests/main/interfaces-password-manager-service/task.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    eval $(printf ubuntu|gnome-keyring-daemon --login)
```
Comment (zyga) [inline tests/main/interfaces-password-manager-service/task.yaml]: The name of the current user depends on the distribution.

File: tests/main/interfaces-password-manager-service/task.yaml [hunk 4]
```diff
@@ -0,0 +1,51 @@
+summary: Ensure that the password-manager-service interface works
+
+systems:
+    - -ubuntu-core-16-*
+
+prepare: |
+    echo "Ensure we have a working gnome-keyring"
+    apt-get install -y --no-install-recommends gnome-keyring
```
Comment (zyga) [inline tests/main/interfaces-password-manager-service/task.yaml]: This needs to be generalized to work on Fedora and openSUSE

File: interfaces/builtin/password_manager_service.go (hunk 1 above)
Comment (mvo5) [inline interfaces/builtin/password_manager_service.go]: Thanks, those are removed now.

File: tests/main/interfaces-password-manager-service/task.yaml (hunk 4 above)
Comment (mvo5) [inline tests/main/interfaces-password-manager-service/task.yaml]: I use distro_install_package now, however I am not sure what the packages are called in fedora/suse that I will need there.

File: tests/main/interfaces-password-manager-service/task.yaml (hunk 2 above)
Comment (mvo5) [inline tests/main/interfaces-password-manager-service/task.yaml]: Yes, nice catch! Thank you!

File: tests/main/interfaces-password-manager-service/task.yaml (hunk 3 above)
Comment (mvo5) [inline tests/main/interfaces-password-manager-service/task.yaml]: indeed, I made it something more neutral.

**Comment by zyga** [comment]
I'll resolve conflicts in this PR later today

**Comment by jdstrand** [comment]
For clarity, when I said I would 'take a look at this' I was saying that I would handle the policy changes and make this work with gnome-keyring and kwallet.

**Comment by jdstrand** [comment]
Per discussion on IRC, I'm going to take over this work and do a new PR. Thanks @mvo5 for your work on this. Closing this PR.

**Pull request closed without merging**